src/Network/Ethereum/Core/BigNumber.js

```diff
@@ -47,9 +47,13 @@ exports.fromStringAsImpl = function (just) {
                 result = new BigNumber(s, radix);
               }
             } else {
-              result = new BigNumber(s, radix);
+                if (s.indexOf('0x') === 0 || s.indexOf('-0x') === 0) {
+                    throw "0x prefix invalid prefix for radix:";
```

Comment:
have you considered returning nothing here?
It's less expected from function like`fromString :: String -> Maybe ___` to throw exceptions

Reply:
Ah i see it's in catch so we would still get nothing, i think it would be better to just `return nothing` here and leave the string as comment maybe.

Reply:
Would be nice to add test case for for this conditional (if there is no already)